src/edu/washington/escience/myria/expression/evaluate/GenericEvaluator.java

```diff
@@ -88,24 +96,26 @@ public void compile() throws DbException {
 
   /**
    * Evaluates the {@link #getJavaExpressionWithAppend()} using the {@link #evaluator}. Prefer to use
-   * {@link #evaluateColumn(TupleBatch)} as it can copy data without evaluating the expression.
+   * {@link #evaluateColumn(TupleBatch)} since it can evaluate an entire TupleBatch at a time for better locality.
    *
    * @param tb a tuple batch
-   * @param rowIdx the row that should be used for input data
-   * @param result the column that the result should be appended to
+   * @param rowIdx index of the row that should be used for input data
+   * @param count column storing number of results
```

Comment:
I'd like to add some check on `count == null` for single value expressions somewhere, or at least a comment on this prerequisite at a proper place. I'm not sure if here is the best place though.